- enable zlib for kmod (so it can deal with gzipped modules)
- disable tests (work around hard-links which are unavailable under AFS)
*** package/kmod/kmod.mk.orig	2016-12-29 12:54:33.000000000 -0800
--- package/kmod/kmod.mk	2017-01-05 14:43:14.938242750 -0800
***************
*** 9,26 ****
  KMOD_SITE = $(BR2_KERNEL_MIRROR)/linux/utils/kernel/kmod
  KMOD_INSTALL_STAGING = YES
  KMOD_DEPENDENCIES = host-pkgconf
! HOST_KMOD_DEPENDENCIES = host-pkgconf
  
  # license info for libkmod only, conditionally add more below
  KMOD_LICENSE = LGPLv2.1+ (library)
  KMOD_LICENSE_FILES = libkmod/COPYING
  
  # static linking not supported, see
  # https://git.kernel.org/cgit/utils/kernel/kmod/kmod.git/commit/?id=b7016153ec8
  KMOD_CONF_OPTS = --disable-static --enable-shared
  
  KMOD_CONF_OPTS += --disable-manpages
! HOST_KMOD_CONF_OPTS = --disable-manpages
  
  ifeq ($(BR2_PACKAGE_ZLIB),y)
  KMOD_DEPENDENCIES += zlib
--- 9,37 ----
  KMOD_SITE = $(BR2_KERNEL_MIRROR)/linux/utils/kernel/kmod
  KMOD_INSTALL_STAGING = YES
  KMOD_DEPENDENCIES = host-pkgconf
! HOST_KMOD_DEPENDENCIES = host-pkgconf host-zlib
  
  # license info for libkmod only, conditionally add more below
  KMOD_LICENSE = LGPLv2.1+ (library)
  KMOD_LICENSE_FILES = libkmod/COPYING
  
+ # T.S: testsuite contains hard links which are not supported under AFS
+ #  argument 1 is the uppercase package name, including an HOST_ prefix
+ #             for host packages
+ define kmod-extract
+       $(INFLATE$(suffix $(KMOD_SOURCE))) $(DL_DIR)/$(KMOD_SOURCE) | \
+       $(TAR) --strip-components=1 --exclude=testsuite -C $($(1)_DIR) $(TAR_OPTIONS) -
+ endef
+ 
+ HOST_KMOD_EXTRACT_CMDS = $(call kmod-extract,HOST_KMOD)
+ KMOD_EXTRACT_CMDS = $(call kmod-extract,KMOD)
+ 
  # static linking not supported, see
  # https://git.kernel.org/cgit/utils/kernel/kmod/kmod.git/commit/?id=b7016153ec8
  KMOD_CONF_OPTS = --disable-static --enable-shared
  
  KMOD_CONF_OPTS += --disable-manpages
! HOST_KMOD_CONF_OPTS = --disable-manpages --with-zlib
  
  ifeq ($(BR2_PACKAGE_ZLIB),y)
  KMOD_DEPENDENCIES += zlib
